awswrangler/neptune/neptune.py

```diff
@@ -0,0 +1,134 @@
+from awswrangler.neptune import client
+from typing import Any
+import pandas as pd
+
+def read_gremlin(
+    client: client,
+    traversal: str
+) -> pd.DataFrame:
+    """Return results of a Gremlin traversal as pandas dataframe.
+
+    Parameters
+    ----------
+    client : neptune.Client
+        instance of the neptune client to use
+    traversal : str
+        The gremlin traversal to execute
+
+    Returns
+    -------
+    Union[pandas.DataFrame, Iterator[pandas.DataFrame]]
+        Results as Pandas DataFrame
+
+    Examples
+    --------
+    Run a Gremlin Query
+
+    >>> import awswrangler as wr
+    >>> client = wr.neptune.Client(host='NEPTUNE-ENDPOINT')
```

Comment:
We don't tend to call a class directly in Wrangler. Instead perhaps we can replicate the pattern [here](https://aws-data-wrangler.readthedocs.io/en/stable/tutorials/030%20-%20Data%20Api.html) where a `connect` method is called and the connection is then reused in other methods?

Reply:
So for Neptune, I intended to use the HTTPS endpoints for all the languages, so there isn't really a long living connection like there is with some other databases.  It's really just a client that has all the needed information to generate the appropriate `requests` call at the time of invocation.  I was thinking that the `client` object here would be very similar to the `connection` object in Redshift.  Thoughts?

Reply:
The client can be implemented as a class, my comment was simply to indicate that no other method in the entire library is calling a class directly and we would prefer to keep it that way for UX. So it can be implemented as a class but there would be a helper to return an instance of the class like you mentioned for [herer](https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/data_api/rds.py#L117)